src/frontend/src/lib/components/dapps/DappCard.svelte

```diff
@@ -2,11 +2,29 @@
 	import DappTags from '$lib/components/dapps/DappTags.svelte';
 	import Logo from '$lib/components/ui/Logo.svelte';
 	import { i18n } from '$lib/stores/i18n.store';
-	import type { OisyDappDescription } from '$lib/types/dapp-description';
+	import type {
+		OisyDappDescription,
+		OisyDappDescriptionDimension
+	} from '$lib/types/dapp-description';
 	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 
 	export let dAppDescription: OisyDappDescription;
 	$: ({ name: dAppName, logo, oneLiner, tags } = dAppDescription);
+
+	let calculateClampValue = (dAppDescription: OisyDappDescriptionDimension) =>
```

Comment:
Please use stricly CSS for styling purpose.

Reply:
It seems unlikely that this can be achieved with plain CSS. While the `text-overflow: ellipsis` feature allows line clamping, it doesn't support multi-line clamping. For this, we need to use the `--webkit-line-clamp: x` property, where x defines the number of lines displayed. However, in our case, the number depends on the number of tags and the title. Hence, it appears that we'll have to calculate it using TypeScript.

Reply:
Thanks for the explanation.

Then we should consider a different approach to this problem. Styling with JavaScript ddoesn’t align with the project’s overall codebase and approach (separation of concerns, maintainability, adherence to best practices, etc.).